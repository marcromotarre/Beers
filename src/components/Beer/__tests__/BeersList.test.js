import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import BeersList from '../BeersList';
import beers from '../../../fixtures/beers';

/** this component should render:
 *  -  a list of beers:
 *          or
 *  - label in casse the list is empty
 *
 */

describe('Beer Component tests', () => {
  it('render Beer Snapshot', () => {
    expect(renderer.create(<BeersList beers={beers} />)).toMatchSnapshot();
    expect(renderer.create(<BeersList beers={[]} />)).toMatchSnapshot();
  });

  it('beers list should render all beers', () => {
    const { getByTestId } = render(<BeersList beers={beers} />);
    const beersListContiner = getByTestId('beers-list-container');
    expect(beersListContiner.children.length).toBe(5);
  });

  it('beers list should render label if there are no beers', () => {
    const { getByTestId } = render(<BeersList beers={[]} />);
    const beersListContiner = getByTestId('beers-list-container');
    expect(beersListContiner.children.length).toBe(1);
    const noBeersLabel = beersListContiner.children[0];
    expect(noBeersLabel.nodeName).toBe('LABEL');
    expect(noBeersLabel.innerHTML).toBe('No beers found');
  });
});
