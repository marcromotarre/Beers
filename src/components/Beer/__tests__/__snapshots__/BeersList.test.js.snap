// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beer Component tests render Beer Snapshot 1`] = `
<section
  className="sc-AxheI hcYtYY"
  data-testid="beers-list-container"
>
  <section
    className="sc-AxjAm bUvDUz"
    data-testid="beer-container"
  >
    <div
      className="sc-AxirZ jpzzvO"
    >
      <div
        className="sc-AxiKw jQbrpK"
      >
        <h1>
          Buzz
        </h1>
        <h2>
          A Real Bitter Experience.
        </h2>
        <p>
          A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.
        </p>
      </div>
      <div
        className="sc-AxhCb elVgfx"
      >
        <label>
          brewed:
        </label>
        <p>
          09/2007
        </p>
      </div>
    </div>
    <div
      className="sc-AxhUy iTCOqd"
    >
      <img
        src="https://images.punkapi.com/v2/keg.png"
      />
    </div>
  </section>
  <section
    className="sc-AxjAm bUvDUz"
    data-testid="beer-container"
  >
    <div
      className="sc-AxirZ jpzzvO"
    >
      <div
        className="sc-AxiKw jQbrpK"
      >
        <h1>
          Trashy Blonde
        </h1>
        <h2>
          You Know You Shouldn't
        </h2>
        <p>
          A titillating, neurotic, peroxide punk of a Pale Ale. Combining attitude, style, substance, and a little bit of low self esteem for good measure; what would your mother say? The seductive lure of the sassy passion fruit hop proves too much to resist. All that is even before we get onto the fact that there are no additives, preservatives, pasteurization or strings attached. All wrapped up with the customary BrewDog bite and imaginative twist.
        </p>
      </div>
      <div
        className="sc-AxhCb elVgfx"
      >
        <label>
          brewed:
        </label>
        <p>
          04/2008
        </p>
      </div>
    </div>
    <div
      className="sc-AxhUy iTCOqd"
    >
      <img
        src="https://images.punkapi.com/v2/2.png"
      />
    </div>
  </section>
  <section
    className="sc-AxjAm bUvDUz"
    data-testid="beer-container"
  >
    <div
      className="sc-AxirZ jpzzvO"
    >
      <div
        className="sc-AxiKw jQbrpK"
      >
        <h1>
          Berliner Weisse With Yuzu - B-Sides
        </h1>
        <h2>
          Japanese Citrus Berliner Weisse.
        </h2>
        <p>
          Japanese citrus fruit intensifies the sour nature of this German classic.
        </p>
      </div>
      <div
        className="sc-AxhCb elVgfx"
      >
        <label>
          brewed:
        </label>
        <p>
          11/2015
        </p>
      </div>
    </div>
    <div
      className="sc-AxhUy iTCOqd"
    >
      <img
        src="https://images.punkapi.com/v2/keg.png"
      />
    </div>
  </section>
  <section
    className="sc-AxjAm bUvDUz"
    data-testid="beer-container"
  >
    <div
      className="sc-AxirZ jpzzvO"
    >
      <div
        className="sc-AxiKw jQbrpK"
      >
        <h1>
          Pilsen Lager
        </h1>
        <h2>
          Unleash the Yeast Series.
        </h2>
        <p>
          Our Unleash the Yeast series was an epic experiment into the differences in aroma and flavour provided by switching up your yeast. We brewed up a wort with a light caramel note and some toasty biscuit flavour, and hopped it with Amarillo and Centennial for a citrusy bitterness. Everything else is down to the yeast. Pilsner yeast ferments with no fruity esters or spicy phenols, although it can add a hint of butterscotch.
        </p>
      </div>
      <div
        className="sc-AxhCb elVgfx"
      >
        <label>
          brewed:
        </label>
        <p>
          09/2013
        </p>
      </div>
    </div>
    <div
      className="sc-AxhUy iTCOqd"
    >
      <img
        src="https://images.punkapi.com/v2/4.png"
      />
    </div>
  </section>
  <section
    className="sc-AxjAm bUvDUz"
    data-testid="beer-container"
  >
    <div
      className="sc-AxirZ jpzzvO"
    >
      <div
        className="sc-AxiKw jQbrpK"
      >
        <h1>
          Avery Brown Dredge
        </h1>
        <h2>
          Bloggers' Imperial Pilsner.
        </h2>
        <p>
          An Imperial Pilsner in collaboration with beer writers. Tradition. Homage. Revolution. We wanted to showcase the awesome backbone of the Czech brewing tradition, the noble Saaz hop, and also tip our hats to the modern beers that rock our world, and the people who make them.
        </p>
      </div>
      <div
        className="sc-AxhCb elVgfx"
      >
        <label>
          brewed:
        </label>
        <p>
          02/2011
        </p>
      </div>
    </div>
    <div
      className="sc-AxhUy iTCOqd"
    >
      <img
        src="https://images.punkapi.com/v2/5.png"
      />
    </div>
  </section>
</section>
`;

exports[`Beer Component tests render Beer Snapshot 2`] = `
<section
  className="sc-AxheI hcYtYY"
  data-testid="beers-list-container"
>
  <label>
    No beers found
  </label>
</section>
`;
